src/components/header.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useContext, useEffect, useRef } from 'react';
+import { MyContext } from './state';
+import { Todo } from '../Types/Todo';
+
+export const Header = () => {
+  const { todos, setTodos, title, setTitle } = useContext(MyContext);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const isAllCompleted = todos.every((todo: Todo) => todo.completed);
+
+  function addTodo(e: React.FormEvent<HTMLFormElement>) {
+    e.preventDefault();
+    const trimmedTitle = title.trim();
+
+    if (trimmedTitle.length === 0) {
+      return;
+    }
+
+    const todo = {
+      title: trimmedTitle,
+      id: Date.now(),
+      completed: false,
+    };
+
+    setTodos((currentTodos: Todo[]) => [...currentTodos, todo]);
+    setTitle('');
+  }
+
+  useEffect(() => {
+    const savedTodos = localStorage.getItem('todos');
+
+    if (savedTodos) {
+      setTodos(JSON.parse(savedTodos));
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    localStorage.setItem('todos', JSON.stringify(todos));
+  }, [todos]);
```

Comment:
why do you need it in the header, move into state.tsx

Reply:
what looks strange?))